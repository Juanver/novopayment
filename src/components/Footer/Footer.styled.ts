import styled from "styled-components";
import { Palette } from "../../assets/palette/Palette";

const ContainerFooter = styled.div`
  background-color: ${Palette.BrandBlue};
`;

export const StyledFooter = {
  ContainerFooter,
};