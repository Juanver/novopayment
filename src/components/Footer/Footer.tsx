import React from 'react'
import { StyledFooter } from './Footer.styled';
import Isotipo from '../../assets/Isotipo.svg';

export const Footer: React.FC = () => {
  const date = new Date();
  const year = date.getFullYear();
  const companyName = 'Novopayment Inc.';

  return (
    <StyledFooter.ContainerFooter>
      <img src={Isotipo} alt='Isotipo' />
      <StyledFooter.ContainerCopyright>
        <small>&copy;{` ${year} `}<span>{companyName} All rights reserved</span></small>
      </StyledFooter.ContainerCopyright>
      <img src={Isotipo} alt='Isotipo' />
    </StyledFooter.ContainerFooter>
  )
}
