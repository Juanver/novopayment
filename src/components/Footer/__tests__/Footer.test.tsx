import { Footer } from "../Footer";
import { render, screen } from "@testing-library/react"

describe('Footer component', () => {
    it('Should render', async () => {
        const { container } = render(<Footer />);
        expect(container).toMatchSnapshot();
    });

    it('Should show the current year', async () => {
        render(<Footer />);
        const text = screen.queryByText(/2023/)
        expect(text).toBeInTheDocument();
    });

    it('Should show the company name', async () => {
        render(<Footer />);
        const text = screen.queryByText(/Novopayment/)
        expect(text).toBeInTheDocument();
    })
})