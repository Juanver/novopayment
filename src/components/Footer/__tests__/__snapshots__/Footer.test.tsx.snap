// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Should render 1`] = `
<div>
  <div
    class="sc-bcXHqe ejgyAO"
  >
    <img
      alt="Isotipo"
    />
    <p
      class="sc-gswNZR dOzeMP"
    >
      <small>
        Â©
         2023 
        <span>
          Novopayment Inc.
           All rights reserved
        </span>
      </small>
    </p>
    <img
      alt="Isotipo"
    />
  </div>
</div>
`;
